src/Utility/getCroppedCanvas.js

```diff
@@ -0,0 +1,78 @@
+export function getCroppedCanvas(ctx) {
+    const {width, height } = ctx.canvas;
+    if (!width && !height) { return ctx.canvas; }
+
+    const imgData = ctx.getImageData(0, 0, width, height);
+    const data = new Uint32Array(imgData.data.buffer);
+    data[0] = 0; // TODO: pointerout handler colors pixel at 0 0 black, so we set it back to transparent;
+                 // I tracked the issue to detectBlockedCanvasImageExtraction function;
+    
+    let shouldBreak = false;
+    let top = null;
+    let bottom = null;
+
+    let idx1 = 0;
+    let idx2 = width * height - 1;
+
+    let x, y;
+    for (y = 0; !shouldBreak && y < height; y++) { 
+        for (x = 0; x < width; x++) {
+            if (!top && data[idx1++]) {
+                top = y + 1;
+                if (bottom) {
+                    shouldBreak = true; 
+                    break; 
+                }
+            }
+            if (!bottom && data[idx2--]) { 
+                bottom = height - y - 1; 
+                if (top) {
+                    shouldBreak = true; 
+                    break;
+                }
+            }
+        }
+
+        if (y > height - y && !top && !bottom) { return ctx.canvas; } // image is blank
+    }
```

Comment:
Have you looked into the performance of that? It seems rather performance heavy and I think I'd rather not crop the canvas than stop the webpage every time a user draws a signature. I was thinking that you might be able to extract the furthest drawing point by looking at the maximum/minimum x and y during drawing. That saves you a lot of time searching for the drawing by looping through the whole image.